cowbird/permissions_synchronizer.py

```diff
@@ -377,7 +402,7 @@ def _filter_used_targets(self, target_res_and_permissions, input_src_res_key, sr
             grp_permissions = handler.get_group_permissions(input_permission.group)
             group_targets = deepcopy(target_res_and_permissions)
 
-        res_data = {}
+        res_data: MutableMapping[str, Tuple[str, List[PermissionResourceData]]] = {}
```

Comment:
In this case, it is really the `PermissionResourceData` because of the extra fields from `PermissionConfigItemType` injected into `ResourceSegment` to form `PermissionResourceData = PermissionConfigItemType | ResourceSegment`.